.github/workflows/manual.yml

```diff
@@ -0,0 +1,36 @@
+on: 
+  workflow_dispatch:
+    inputs:
+      engine_sha:
+        description: 'engine hash'     
+        required: true
+      branch:
+        description: 'branch to be cherry picked onto'     
+        required: false
+        default: 'master'
+
+jobs:
+  roll-engine-to-framework:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Flutter Release
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.branch }}
+          repository: flutter/flutter
+          token: ${{ github.token }}
+      - name: Update Engine version
+        run: |
+          echo "${{ github.event.inputs.engine_sha }}" > bin/internal/engine.version
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v5
+        with:
+          token: ${{ secrets.PAT }}
+          commit-message: Roll engine to ${{ github.event.inputs.engine_sha }}
+          committer: GitHub <noreply@github.com>
+          reviewers: itsjustkevin
+          title: 'Roll engine to ${{ github.event.inputs.engine_sha }}'
+          body: |
+            flutter engine->framework roll
```

Comment:
Is it possible to have this be the git log between the previous commit and the new revision?

Reply:
Finally get commit history to work.

This took so many hours because the [marketplace action](https://github.com/marketplace/actions/create-pull-request) imposes so many constraints:
1. [marketplace create pr](https://github.com/marketplace/actions/create-pull-request) doesn't support excluding files. therefore the only way to pass down the output of git log is to use env variable, and the variable format cannot be processed by `sed` or `tr`, requires [multiline env variable syntax](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings) 
2. [marketplace checkout](https://github.com/marketplace/actions/create-pull-request) has a `fetch-depth` field that needs to be set for git history to work. By default this value is 1, which only has the history of the most recent commit.
3. [marketplace create pr](https://github.com/marketplace/actions/create-pull-request) only creates pr to the currently checked out branch. Since we checkout framework first and then engine to get commit history, we had to set working directory and pr/checkout paths for each step.

(umm personally i feel I could have saved a lot of time debugging and circumventing, if we don't rely on marketplace actions)